espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/GetDisplayed.java

```diff
@@ -30,11 +32,10 @@
 
     @Override
     public Boolean handle(AppiumParams params) throws AppiumException {
-        final ViewInteraction viewInteraction = Element.getById(params.getElementId());
         try {
-            viewInteraction.check(matches(isDisplayed()));
+            Element.getById(params.getElementId());
```

Comment:
why so?

Reply:
Because `Element.getById` only returns an element if the `viewInteraction.check(matches(isDisplayed()));` has passed. Otherwise it throws a `StaleElementException`.

Reply:
does it always have an id?

Reply:
As much as it ever did. If it doesn't `Element.getById` will throw a `NoSuchElementException` as it did before.